src/averageAges.js

```diff
@@ -55,7 +76,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const relations = {};
+  const women = people.filter(person => person.sex === 'f');
+
+  women.forEach(woman => {
+    relations[woman.name] = { born: woman.born };
+  });
+
+  let dateCounter = 0;
+  let count = 0;
+
+  for (const person of people) {
+    if (onlyWithSon && person.sex !== 'm') {
+      continue;
+    }
+
+    if (person.mother in relations) {
+      dateCounter += (person.born - relations[person.mother].born);
+      count++;
+    }
+  }
```

Comment:
https://mateacademy.slack.com/archives/G01KEC0048K/p1615560295107800

Reply:
changed all methods, but could not do completely without if